seqr/views/apis/anvil_workspace_api.py

```diff
@@ -0,0 +1,118 @@
+    :param name: The name of the workspace. It also be used as the project name
+    :return the projectsByGuid with the new project json
+    """
+    # Validate that the current user has logged in through google and has one of the valid can_edit levels of
+    #  access on the specified workspace
+    project = Project.objects.filter(name = name)
+    if project:
+        error = 'Project {} exists.'.format(name)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    if not _workspace_can_edit(request.user, namespace, name):
+        return create_json_response({
+            'projectsByGuid': {},
+        })
+
+    # Validate all the user inputs from the post body
+    request_json = json.loads(request.body)
+
+    missing_fields = [field for field in ['genomeVersion', 'uploadedFileId'] if not request_json.get(field)]
+    if missing_fields:
+        error = 'Field(s) "{}" are required'.format(', '.join(missing_fields))
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    if not request_json.get('agreeSeqrAccess'):
+        error = 'Must agree to grant seqr access to the data in the associated workspace.'
+        return create_json_response({'error': error}, status = 400, reason = error)
+
+    # Add the seqr service account to the corresponding AnVIL workspace
+    if add_service_account(request.user, namespace, name):
+        # Create a new Project in seqr
+        project_args = {
+            'name': name,
+            'genome_version': request_json['genomeVersion'],
+            'description': request_json.get('description', ''),
+            'workspace_namespace': namespace,
+            'workspace_name': name,
+        }
+
+        project = create_model_from_json(Project, project_args, user = request.user)
+
+        # Add families/individuals based on the uploaded pedigree file
+        try:
+            json_records = load_uploaded_file(request_json['uploadedFileId'])
+        except Exception as ee:
+            error = "Uploaded pedigree file is missing or other exception: {}".format(str(ee))
+            return create_json_response({'error': error}, status = 400, reason = error)
+
+        # todo: update families and individuals according to the uploaded individual records
+        updated_families, updated_individuals = _add_or_update_individuals_and_families(
+            project, individual_records = json_records, user = request.user
+        )
+
+        # todo: Send an email to all seqr data managers saying a new AnVIL project is ready for loading. Include the seqr
+        # project guid, the workspace name, and attach a txt file with a list of the individual IDs that were created.
+        return create_json_response({'projectGuid':  project.guid})
+
+    error = 'Failed to grant seqr service account access to the workspace'
+    return create_json_response({'error': error}, status = 400, reason = error)
```

Comment:
I know this is super nitpicky, but we don't put spaces around equals signs, and it makes the code style inconsistent which isn't great for long term maintainability. This is probably a setting you can change in whatever IDE/ text editor you use for writing code